src/components/SideBar/ExpandedPostSidebar.tsx

```diff
@@ -0,0 +1,243 @@
+
+const styles = {
+  ad: {
+    height: "100%",
+    width: "100%",
+  },
+  adSpaceContainer: {
+    alignItems: "center",
+    backgroundColor: "gray.light",
+    border: "5px solid",
+    borderColor: "primary.main",
+    borderRadius: 5,
+    boxSizing: "border-box",
+    cursor: "pointer",
+    display: "flex",
+    height: "30%",
+    justifyContent: "center",
+    marginTop: 0,
+    overflow: "hidden",
+    width: "100%",
+  },
+  adTitle: {
+    color: "primary.main",
+    fontWeight: "bold",
+    textAlign: "center",
+  },
+  followButton: {
+    boxShadow: "none",
+    "&:hover": {
+      boxShadow: "none",
+      backgroundColor: "primary.main",
+    },
+  },
+  footerContainer: {
+    height: "5%",
+    width: "100%",
+  },
+  footerItem: {
+    color: "gray.main",
+    paddingRight: "0.75rem",
+  },
+  footerItemText: {
+    fontSize: "0.75rem",
+  },
+  footerNav: {
+    display: "flex",
+    flexWrap: "wrap",
+    height: "100%",
+    paddingX: 2,
+  },
+  headerTitle: {
+    fontWeight: "bold",
+    paddingX: 2,
+    paddingTop: 1,
+  },
+  relevantUserContainer: {
+    border: "2px solid",
+    borderColor: "gray.light",
+    borderRadius: 5,
+    display: "flex",
+    flexDirection: "column",
+    height: "14%",
+    width: "100%",
+    overflow: "hidden",
+  },
+  rightContent: {
+    alignItems: "center",
+    display: "flex",
+    flexDirection: "column",
+    gap: 2,
+    height: "100vh",
+    paddingLeft: "40px",
+    position: "fixed",
+    width: "310px",
+  },
+  ruContent: {
+    display: "flex",
+    flexDirection: "row",
+    paddingX: 2,
+    paddingY: 1,
+    alignItems: "center",
+    justifyContent: "space-between",
+    backgroundColor: "transparent",
+    "&:hover": {
+      backgroundColor: "#E9EBED",
+    },
+    border: "none",
+    cursor: "pointer",
+  },
+  ruDisplayName: {
+    fontWeight: "bold",
+    fontSize: "1rem",
+  },
+  ruHeader: {
+    width: "100%",
+  },
+  ruNameLink: {
+    color: "black",
+  },
+  ruNames: { marginLeft: 2, textAlign: "left", marginY: "6px" },
+  ruTitle: {
+    fontWeight: "bold",
+    paddingX: 2,
+    paddingTop: 1,
+  },
+  ruUsername: {
+    color: "rgba(0, 0, 0, 0.6)",
+  },
+  searchBarContainer: {
+    height: "5%",
+    width: "100%",
+  },
+  userInfo: {
+    display: "flex",
+    alignItems: "center",
+  },
+};
+
+const ExpandedPostSidebar = () => {
+  const relevantUser = useAppSelector((state) => state.expandedPost);
+  return (
+    <Box sx={styles.rightContent}>
+      <Box sx={styles.searchBarContainer}>
+        <SearchBar placeholder="Search Chirp" />
+      </Box>
+
+      <Box sx={styles.relevantUserContainer}>
+        <Box sx={styles.ruHeader}>
+          <Typography variant="h6" sx={styles.ruTitle}>
+            Relevant People
+          </Typography>
+        </Box>
+        <Box sx={styles.ruContent} className="ruContent">
+          <Box sx={styles.userInfo}>
+            <Avatar />
+            <Box sx={styles.ruNames}>
+              <Link
```

Comment:
Instead of using `href`, try passing a react-router-dom `Link` to the `component` prop and adding the route to the `to` prop. You can look at some of our other usages of links or https://mui.com/material-ui/guides/routing/ for examples. This applies to other usages of links for redirects too

Reply:
Applied this and removed any unnecessary `Link` calls as well